syntax = "proto3";

package user_service;

// User message definition
message User {
  int32 id = 1;
  string name = 2;
  string email = 3;
}

// Request messages
message GetUserRequest {
  int32 id = 1;
}

message CreateUserRequest {
  string name = 1;
  int32 id = 2;
  string email = 3;
}

message UpdateUserRequest {
  int32 id = 1;
  string name = 2;
  string email = 3;
}

message DeleteUserRequest {
  int32 id = 1;
}

message GetAllUsersRequest {
  // Empty request for getting all users
}

// Response messages
message GetUserResponse {
  bool success = 1;
  string message = 2;
  User user = 3;
}

message CreateUserResponse {
  bool success = 1;
  string message = 2;
  User user = 3;
}

message UpdateUserResponse {
  bool success = 1;
  string message = 2;
  User user = 3;
}

message DeleteUserResponse {
  bool success = 1;
  string message = 2;
  User user = 3;
}

message GetAllUsersResponse {
  bool success = 1;
  string message = 2;
  repeated User users = 3;
}

// Service definition
service UserService {
  rpc GetUser(GetUserRequest) returns (GetUserResponse);
  rpc CreateUser(CreateUserRequest) returns (CreateUserResponse);
  rpc UpdateUser(UpdateUserRequest) returns (UpdateUserResponse);
  rpc DeleteUser(DeleteUserRequest) returns (DeleteUserResponse);
  rpc GetAllUsers(GetAllUsersRequest) returns (GetAllUsersResponse);
}